sdk/dnsresolver/Microsoft.Azure.Management.DnsResolver/tests/ScenarioTests/InboundEndpointTests.cs

```diff
@@ -0,0 +1,567 @@
+            listedInboundEndpoints.All(inboundEndpoint => ValidateInboundEndpointIsExpected(inboundEndpoint, expectedInboundEndpoints));
+        }
+
+        [Fact]
+        public void ListInboundEndpointsInResourceGroup_ListNextPage_ExpectNextEndpointsRetrieved()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var createdDnsResolver = this.CreateDnsResolver(resourceGroupName: resourceGroupName);
+            var numInboundEndpoints = 2;
+            var top = 1;
+            var createdInboundEndpoints = this.CreateInboundEndpoints(createdDnsResolver, resourceGroupName: resourceGroupName, numInboundEndpoints: numInboundEndpoints);
+            var expectedInboundEndpoints = createdInboundEndpoints.OrderBy(x => x.Name).Skip(top);
+            var initialListResult = this.DnsResolverManagementClient.InboundEndpoints.List(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name,
+                top: top);
+
+
+            var nextListResult = this.DnsResolverManagementClient.InboundEndpoints.ListNext(
+                initialListResult.NextPageLink);
+
+            var listedInboundEndpoints = nextListResult.ToArray();
+            listedInboundEndpoints.Count().Should().Be(numInboundEndpoints - top);
+            listedInboundEndpoints.All(inboundEndpoint => ValidateInboundEndpointIsExpected(inboundEndpoint, expectedInboundEndpoints));
+        }
+
+        [Fact]
+        public void ListInboundEndpointsInResourceGroup_NoEndpointPresents_ExpectNoEndpointRetrieved()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var createdDnsResolver = this.CreateDnsResolver(resourceGroupName: resourceGroupName);
+
+            var listResult = this.DnsResolverManagementClient.InboundEndpoints.List(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name);
+
+            listResult.Should().NotBeNull();
+            listResult.Count().Should().Be(0);
+        }
+
+        [Fact]
+        public void DeleteInboundEndpoint_EndpointExists_ExpectEndpointDeleted()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var createdDnsResolver = this.CreateDnsResolver(resourceGroupName: resourceGroupName);
+            var creatednboundEndpoint = this.CreateInboundEndpoint(createdDnsResolver: createdDnsResolver, resourceGroupName: resourceGroupName);
+
+            this.DnsResolverManagementClient.InboundEndpoints.Delete(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name,
+                inboundEndpointName: creatednboundEndpoint.Name);
+
+            var listResult = this.DnsResolverManagementClient.InboundEndpoints.List(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name);
```

Comment:
Why use list and not GET?

Reply:
I wanted to ensure there is no side effect from delete Dns Resolver.

Reply:
You're deleting inbound endpoint. What side effect do you expect?

Reply:
May be the Get the IE shows NoFound but List IEs show not updated?

Reply:
That seems improbable. Remember you're testing DELETE here, not the GET API.

Reply:
GET and LIST makes no difference in terms of validation outcome. 
If you prefer GET here, I can make change here.

Reply:
Yes please, let's use GET.

As you said GET vs. LIST doesn't make a difference in outcome but GET is faster and GET is more intuitive to test deletion result.